.github/workflows/codeql-analysis.yml

```
@@ -0,0 +1,32 @@
+name: "CodeQL"
+# see https://codeql.github.com/docs/
+
+on:
+  push:
+    branches: [master]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [master]
+
+jobs:
+  analyse:
+    name: Analyse
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
+      with:
+        # We must fetch at least the immediate parents so that if this is
+        # a pull request then we can checkout the head.
+        fetch-depth: 2
+
+    - name: Initialize CodeQL
+      uses: github/codeql-action/init@v2
+
+    - run: ./bootstrap.sh
+    - run: ./configure
```

Similar to another change request, let's expose more of our code to CodeQL analysis:

```suggestion
    - run: ./configure --with-openssl
```

Alternatively, we can (and probably even should) build with all the maximus layer options, but that requires more work. The above (simple) suggestion is already tested by default.yaml.